src/redis_pubsub.rs

```diff
@@ -0,0 +1,165 @@
+use redis::{Client, Commands, RedisError};
+use std::cell::RefCell;
+use std::collections::HashMap;
+use std::thread;
+use std::time::Duration;
+
+static ERROR_CHANNEL: &'static str = "RUSTG_REDIS_ERROR_CHANNEL";
+
+thread_local! {
+    static REQUEST_SENDER: RefCell<Option<flume::Sender<PubSubRequest>>> = RefCell::new(None);
+    static RESPONSE_RECEIVER: RefCell<Option<flume::Receiver<PubSubResponse>>> = RefCell::new(None);
+}
+
+enum PubSubRequest {
+    Subscribe(String),
+    Publish(String, String),
+}
+
+// response might not be a good name, since those are not sent in response to requests
+enum PubSubResponse {
+    Disconnected(String),
+    Message(String, String),
+}
+
+fn handle_redis_inner(
+    client: Client,
+    control: &flume::Receiver<PubSubRequest>,
+    out: &flume::Sender<PubSubResponse>,
+) -> Result<(), RedisError> {
+    let mut conn = client.get_connection()?;
+    let mut pub_conn = client.get_connection()?;
+    let mut pubsub = conn.as_pubsub();
+    let _ = pubsub.set_read_timeout(Some(Duration::from_secs(1)));
```

Comment:
I'm assuming this is another `Result` `must_use`, what's the error states of this and why are we avoiding it?